components/remote_settings/src/remote_settings.udl

```diff
@@ -69,7 +71,53 @@ interface RemoteSettings {
     [Throws=RemoteSettingsError]
     RemoteSettingsResponse get_records_since(u64 timestamp);
 
+    // Get records from the cache, if present
+    //
+    // This method will only check the cache and not make any network requests.
+    // If a cache was not setup, this will always return null.
+    RemoteSettingsResponse? get_cached_records();
+
+    // Sync records with the cache
+    //
+    // This method fetches new records from the server and updates the cache.
+    // If a cache was not setup, this is a no-op
+    [Throws=RemoteSettingsError]
+    void sync_cached_records();
+
     // Download an attachment with the provided id to the provided path.
     [Throws=RemoteSettingsError]
     void download_attachment_to_path(string attachment_id, string path);
 };
+
+// Builder object for constructing a RemoteSettings instance
+interface RemoteSettingsBuilder {
+    constructor();
+
+    [Self=ByArc]
+    RemoteSettingsBuilder collection_name(string collection_name);
+
+    [Self=ByArc]
+    RemoteSettingsBuilder server(RemoteSettingsServer server);
+
+    [Self=ByArc]
+    RemoteSettingsBuilder bucket_name(string bucket_name);
+
+    // Setup a cache that writes to a file.
+    // This is the simplest way to enable caching support.
+    [Self=ByArc, Throws=RemoteSettingsError]
+    RemoteSettingsBuilder cache_file(string path);
```

Comment:
Let's also note the consumers that it's up to them to ensure the path is valid and writable, otherwise, the `cache.store()` will fail and it doesn't support error handling at that point.